ddl/ddl_api.go

```diff
@@ -5245,48 +5245,66 @@ func checkPlacementSpecConstraint(rules []*placement.Rule, rule *placement.Rule,
 			return rules, err
 		}
 
+		rulesLen := len(rules)
+		ruleCnt := rule.Count
 		for labels, cnt := range constraints {
-			newRule := &placement.Rule{}
+			newRule := &placement.RuleOp{}
 			*newRule = *rule
+			newRule.Rule = &placement.Rule{}
+			*newRule.Rule = *rule.Rule
```

Comment:
How about create a `clone` method for `RuleOp`?

Reply:
It is not reusing code. But it can clarify the intention. Sounds good.